Plugin uploads: Disable for sites on multisite networks

seear opened the pull request:
The current jetpack theme and plugin upload APIs do [not currently work](https://github.com/Automattic/jetpack/pull/6201) for Jetpack sites on multisite networks, so disable the plugin upload feature for now on those sites.

<img width="1247" alt="screen shot 2017-07-31 at 16 10 15" src="https://user-images.githubusercontent.com/7767559/28784413-4c6fcf3c-760b-11e7-88ef-a4ab84fd68dc.png">

**To Test**
* Checkout this branch or use the calypso.live link
* Go to /plugins/upload/{jetpack_sites}
* For sites on a multisite network you should see a message as in the screenshot above
* Regular jetpack sites should allow uploads as usual
* If you don't have a multisite install available for testing, modify the code so that the new `isJetpackMultisite` prop is always true

seear commented:
Going to tweak this a bit after rebase...

seear commented:
> Going to tweak this a bit after rebase...

Ok, now we don't bother with the jetpack upgrade message for multisite installs.

kristastevens commented:
I wonder, since this messaging is a function of a gap we currently have in Calypso, that we simply say on the first line: "Visit WP Admin to install your plugin." 

The button text then says "Go to WP Admin" and opens the plugins installation screen on click. 

My thinking here is that the user doesn't care why they can't upload a plugin here on this particular screen -- they just want to accomplish their goal of uploading a plugin, so we should make that as clean and simple as possible.

seear commented:
Excellent, thanks @kristastevens!

I've updated the wording and changed the button to link to wp-admin/plugin-install.php:
<img width="823" alt="screen shot 2017-08-01 at 13 19 02" src="https://user-images.githubusercontent.com/7767559/28825083-b8377b64-76bc-11e7-88b2-830894a269a3.png">

Button goes to:
<img width="654" alt="screen shot 2017-08-01 at 13 21 03" src="https://user-images.githubusercontent.com/7767559/28825073-aee43af2-76bc-11e7-8052-08ed3811af00.png">

kristastevens commented:
Looking good, @seear! I think we can even take out "Not available on multisite networks" -- the function is available, just in a different screen.

seear commented:
> I think we can even take out "Not available on multisite networks" -- the function is available, just in a different screen.

I'm not certain on this. We don't yet have the _Upload plugin_ buttons that link to this page, but when we do we can make them go straight to wp-admin for multisite sites, so the user should never see this screen.

People that do end up here would typically have multiple sites (maybe they're on this screen for one site and they use the site picker to select a multisite site), so it may still be useful to indicate why upload is not available.

kristastevens commented:
I think it's of limited use, though if you feel strongly, I defer.

seear commented:
I could go either way, but I think it is a better product if we don't have to mention multisite, so I've taken it out:
<img width="789" alt="screen shot 2017-08-01 at 14 02 24" src="https://user-images.githubusercontent.com/7767559/28826536-60d95cec-76c2-11e7-8c0e-5f0dbb1dc0e1.png">

seear commented:
thanks again @kristastevens, I think that is much improved from where we started.

kristastevens commented:
Thanks so much for giving Editori al a ping! Really happy to have a chance to help out.

tyxla commented:
Looks really awesome, thanks so much @kristastevens and @seear!
